### ISSUE TITLE
define collections to expose to api

### ISSUE THREAD

**Issue by mario-minati** [preaction/Yancy#34]
Hello Doug,

on using dbi backend, we'd like to define the collections to expose to the API (instead using x-ignore to define which to not expose). We'd like adding a mode switch to allow switching between expose-on-default or ignore-on-default. 

Would that be an acceptable extension?

Best greets,

Mario

**Comment by preaction** [comment]
DBIx::Class I expect? If you don't use the `read_schema` flag, then you can define your own collections your own self. But, I think you might think that's a lot of work, and I can respect that (and agree wholeheartedly that it certainly causes a bunch of unwelcome duplication). 

So, what if I made it so you could specify only the collections you want to `read_schema`? Like: `read_schema: [ 'collection', ... ]` perhaps? Then only those collections would be read from the backend and exposed to the editor/API/etc...

**Comment by mario-minati** [comment]
Yes DBIx::Class, and we use `read_schema`. First I copied parts of your code, but I don't like that way, as we wan't to participate from further developments and give back our extensions.

As our currently only internally used application will be developed further and further we need to make sure that future db tables will not be accessible via the OpenAPI interface unintentionally, that's the cause of the idea.

Making read_schema a list is a nice idea, but I'd rather use the existing collection hash structure or if I may dream, make it configurable in DBIx::Class result classes. If we could make the current collection options be picked from the result classes we can configure database table accessibility in a single place. :-)

**Comment by preaction** [comment]
The existing collection hash structure would make it a bit more verbose:

```
collections => {
    foo => { read_schema => 1 },
    bar => { read_schema => 1 },
    baz => { read_schema => 1 },
    fizz => { read_schema => 1 },
    buzz => { read_schema => 1 },
}
```

I'm not against that, indeed that was my first thought, but it's longer than:

```
read_schema => [qw( foo bar baz fizz buzz )],
```

Adding `read_schema` flags to the existing `collections` hash instead seems like a fine idea. Also, if we're assuming that other things will always need to be added to the collection's configuration (which we probably should, since the default list columns are almost always wrong), then I'd be more inclined to the collection-level `read_schema` flag.

As for adding a `yancy` configuration item to the DBIx::Class result class: I'm not sure how one goes about adding result-level metadata. I know one can add column-level metadata just by putting it in there. But for result-level metadata there'd need to be some kind of DBIx::Class::Yancy class to be able to add the configuration (and a flag in the Yancy::Backend::Dbic to restrict it to using those result classes that have that component). I'd be perfectly happy with something like that: Perhaps a check for a `yancy` subroutine on the result class that returns the Yancy configuration for that result class.

**Comment by mario-minati** [comment]
Using a yancy subroutine and testing it with `can` would be working, but is not likely elegant. A DBIx::Class::Yancy could provide more syntactically sugar. 

Is `collections => { foo => { read_schema => 1} }` the correct naming in terms of the desired renaming in v2.0 of Yancy, as you plan to rename `collection` to `schema` if I got it right?
